ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java

```diff
@@ -222,4 +222,50 @@ private BackupReplayFile newReplayFile() {
         ? Optional.of(new BackupReplayFile(latestBakFile))
         : Optional.empty();
   }
+
+  private void ensureDirectoryExists(final File backupsDir) {
+    if (!backupsDir.exists()) {
+      if (!backupsDir.mkdirs()) {
+        throw new KsqlServerException("Couldn't create the backups directory: "
+            + backupsDir.getPath()
+            + "\n Make sure the directory exists and is readable/writable for KSQL server "
+            + "\n or its parent directory is readable/writable by KSQL server"
+            + "\n or change it to a readable/writable directory by setting '"
+            + KsqlConfig.KSQL_METASTORE_BACKUP_LOCATION
+            + "' config in the properties file."
+        );
+      }
+
+      try {
+        Files.setPosixFilePermissions(backupsDir.toPath(),
```

Comment:
Does it make sense to explicitly set permissions here? I see we don't do that for the Kafka Streams state directory. I'm in favor of consistency unless there's a good reason to deviate.

Reply:
I'd like to keep the metastore directory restricted from other uses to access the data. By default, the directory created has `drwxrwxr-x` which is opened to any user in the system. It's up to the user to specify a directory that has restricted permissions only to the required users, but if KSQL creates it automatically, then I set those here too.

I don't know why the streams state directory is not protected. We might want to do that. I won't do it in this patch until I understand their reason.